packages/flutter_test/lib/src/binding.dart

```diff
@@ -920,6 +921,13 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     // So that we can assert that it remains the same after the test finishes.
     _beforeTestCheckIntrinsicSizes = debugCheckIntrinsicSizes;
 
+    bool shouldTearDownVerifyInvariants = false;
+    addTearDown(() {
+      if (shouldTearDownVerifyInvariants) {
+        _verifyTearDownInvariants();
+      }
+    });
+
```

Comment:
I'm concerned about the addition of the local flag that's referenced in the teardown if for no other reason than readability.

Since `addTearDown` can be called at any point in the method, could we move it to line ~961 where the setting of `shouldTearDownVerifyInvariants` is happening now? Then there won't be a need for a local flag, and the call to `addTearDown` can also be simplified (since it wouldn't need to check the flag).

Reply:
I am worried not. This is because `addTearDown`'s ordering is important. For example, users can call addTearDown inside their testBody (e.g. me! so I guess a lot of people). Then this refactor will fail

Reply:
Shoot, yeah, you're right. Ordering is going to be very important in this case, so we _can't_ just do the `addTearDown` anywhere... Thanks!

Reply:
You are welcome!